File: functions/Petshop.Shared/Models/Agendamento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Petshop.Shared.Enums;

namespace Petshop.Shared.Models;

[Table("agendamentos")]
public class Agendamento
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required(ErrorMessage = "Data do agendamento é obrigatória")]
    [Column("data_agendamento", TypeName = "DATE")]
    public DateTime DataAgendamento { get; set; }

    [Required(ErrorMessage = "Horário é obrigatório")]
    [Column("horario")]
    public TimeSpan Horario { get; set; }

    [Required(ErrorMessage = "Método de atendimento é obrigatório")]
    [StringLength(20)]
    [Column("metodo_atendimento")]
    public string MetodoAtendimento { get; set; } = string.Empty; // telebusca, local

    [StringLength(20)]
    [Column("porte_pet")]
    public string? PortePet { get; set; } // pequeno, medio, grande

    [StringLength(500)]
    [Column("observacoes")]
    public string? Observacoes { get; set; }

    [Required(ErrorMessage = "Valor total é obrigatório")]
    [Range(0.01, double.MaxValue, ErrorMessage = "Valor deve ser maior que zero")]
    [Column("valor_total")]
    public double ValorTotal { get; set; }

    [Column("status")]
    public StatusAgendamento Status { get; set; } = StatusAgendamento.Pendente;

    // Relacionamentos
    [Required]
    [Column("cliente_id")]
    public long ClienteId { get; set; }

    [ForeignKey("ClienteId")]
    public Cliente? Cliente { get; set; }

    [Required]
    [Column("pet_id")]
    public long PetId { get; set; }

    [ForeignKey("PetId")]
    public Pet? Pet { get; set; }

    // Relacionamento Many-to-Many com Servicos
    public ICollection<Servico> Servicos { get; set; } = new List<Servico>();
}

File: functions-java/petshop-functions-shared/src/main/java/com/petshop/functions/shared/model/Agendamento.java
package com.petshop.functions.shared.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "agendamentos")
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Data do agendamento é obrigatória")
    @Column(name = "data_agendamento", nullable = false)
    private LocalDate dataAgendamento;

    @NotNull(message = "Horário é obrigatório")
    @Column(nullable = false)
    private LocalTime horario;

    @NotBlank(message = "Método de atendimento é obrigatório")
    @Column(name = "metodo_atendimento", nullable = false, length = 20)
    private String metodoAtendimento; // telebusca, local

    @Column(name = "porte_pet", length = 20)
    private String portePet; // pequeno, medio, grande

    @Column(length = 500)
    private String observacoes;

    @NotNull(message = "Valor total é obrigatório")
    @DecimalMin(value = "0.0", inclusive = false, message = "Valor deve ser maior que zero")
    @Column(name = "valor_total", nullable = false)
    private Double valorTotal;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StatusAgendamento status = StatusAgendamento.PENDENTE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false)
    private Pet pet;

    @ManyToMany
    @JoinTable(
        name = "agendamento_servicos",
        joinColumns = @JoinColumn(name = "agendamento_id"),
        inverseJoinColumns = @JoinColumn(name = "servico_id")
    )
    private List<Servico> servicos = new ArrayList<>();

    // Enum para status
    public enum StatusAgendamento {
        PENDENTE,
        CONFIRMADO,
        EM_ANDAMENTO,
        CONCLUIDO,
        CANCELADO
    }

    // Construtores
    public Agendamento() {
    }

    public Agendamento(LocalDate dataAgendamento, LocalTime horario, String metodoAtendimento, 
                       String portePet, String observacoes, Double valorTotal) {
        this.dataAgendamento = dataAgendamento;
        this.horario = horario;
        this.metodoAtendimento = metodoAtendimento;
        this.portePet = portePet;
        this.observacoes = observacoes;
        this.valorTotal = valorTotal;
    }

    // Getters e Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getDataAgendamento() {
        return dataAgendamento;
    }

    public void setDataAgendamento(LocalDate dataAgendamento) {
        this.dataAgendamento = dataAgendamento;
    }

    public LocalTime getHorario() {
        return horario;
    }

    public void setHorario(LocalTime horario) {
        this.horario = horario;
    }

    public String getMetodoAtendimento() {
        return metodoAtendimento;
    }

    public void setMetodoAtendimento(String metodoAtendimento) {
        this.metodoAtendimento = metodoAtendimento;
    }

    public String getPortePet() {
        return portePet;
    }

    public void setPortePet(String portePet) {
        this.portePet = portePet;
    }

    public String getObservacoes() {
        return observacoes;
    }

    public void setObservacoes(String observacoes) {
        this.observacoes = observacoes;
    }

    public Double getValorTotal() {
        return valorTotal;
    }

    public void setValorTotal(Double valorTotal) {
        this.valorTotal = valorTotal;
    }

    public StatusAgendamento getStatus() {
        return status;
    }

    public void setStatus(StatusAgendamento status) {
        this.status = status;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Pet getPet() {
        return pet;
    }

    public void setPet(Pet pet) {
        this.pet = pet;
    }

    public List<Servico> getServicos() {
        return servicos;
    }

    public void setServicos(List<Servico> servicos) {
        this.servicos = servicos;
    }

    // Métodos auxiliares
    public void adicionarServico(Servico servico) {
        servicos.add(servico);
        servico.getAgendamentos().add(this);
    }

    public void removerServico(Servico servico) {
        servicos.remove(servico);
        servico.getAgendamentos().remove(this);
    }
}
